### DISCUSSION TITLE
when i train with my dataset,fedot usually fits it with weak model.

### DISCUSSION THREAD

**Discussion by guess-guess-who-i-am** [nccr-itmo/FEDOT#1401] [Q&A]
thank you for your reading. when i train with my dataset,fedot usually fits it with weak model.i think maybe i am not really familiar with fedot, so please help me.

my dataset:https://www.kaggle.com/datasets/rakannimer/air-passengers

graph structure:

<img width="700" height="700" alt="Figure_1" src="https://github.com/user-attachments/assets/1ded6913-595f-41d6-83d1-74425e561e73" />

final result（The number on the x-axis are huge because i use pd.to_numeric(df['Month'],downcast='float') to turn it into the form that fedot can process):

<img width="640" height="480" alt="Figure_2" src="https://github.com/user-attachments/assets/e457c0a8-6fc1-4b57-a440-25bde6ddcda3" />

my code :
`
from fedot.api.main import Fedot
import pandas as pd
from fedot.core.repository.tasks import TsForecastingParams, Task, TaskTypesEnum
import matplotlib.pyplot as plt
import matplotlib
matplotlib.use('TkAgg')
df = pd.read_csv(r"C:\Users\23859\Desktop\AirPassengers.csv")
df['Month'] = pd.to_datetime(df['Month'])#Transform time series into manageable time series
df['Month'] = pd.to_numeric(df['Month'],downcast='float')
size = int(len(df)*0.7)
X_train = df.iloc[:size].values
X_test = df.iloc[size:].values
model = Fedot(problem='ts_forecasting',
              task_params=Task(TaskTypesEnum.ts_forecasting,
                               TsForecastingParams(forecast_length=X_test.shape[0])).task_params,
              timeout=60,
              n_jobs=-1,
              metric='mae')
model.fit(features=X_train[:,0],target=X_train[:,1])
y_pred = model.forecast(pre_history=X_test[:,0])
model.current_pipeline.show()
plt.plot(X_train[:,0], X_train[:,1], label='features')
plt.plot(X_test[:,0], X_test[:,1], label='target')
plt.plot(X_test[:,0], y_pred, label='fedot')
plt.grid()
plt.legend()
plt.show()

`

actually the problem can be solved by arima perfectly. and i find out that arima is in the operation list.  so i don't know why fedot do not use arima instead.

arima result(from :https://www.youtube.com/watch?v=TSfdvQ0ARwI)
<img width="991" height="634" alt="image" src="https://github.com/user-attachments/assets/d2249f43-268e-4143-b941-40415b987a99" />

please give me a hand. thank you very much.

**Comment by nicl-nno** [comment]
Hello!

In the past, we faced some issues with using ARIMA-like methods as a candidate model for optimiser. We will re-check it and answer with details and possible fixes.

**Reply by guess-guess-who-i-am** [reply]
hello!

thank you for your reply and thank you for your open-source. i have read the code of fedot and golem. i think the reason is that the original params of arima makes it can not fit the data perfectly, so golem choose another pipeline that can fit the data better. then golem create more individuals by mutation or crossover. 

theoretically，fedot can return the best pipeline. but due to the limit of time or iteration,  arima may not be used again.  maybe i can apply great params for arima then use fedot to improve the performance.May I ask if you have any other good methods？

thank you very much.
